datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -455,7 +460,13 @@ def should_bypass_proxy(url, no_proxy_uris):
             #   e.g. "foo.com" matches "foo.com" and "bar.foo.com"
             # A domain name with a leading "." matches subdomains only.
             #   e.g. ".y.com" matches "x.y.com" but not "y.com".
-            dot_no_proxy_uri = no_proxy_uri if no_proxy_uri.startswith(".") else ".{}".format(no_proxy_uri)
+            if no_proxy_uri.startswith((".", "*.")):
+                # Support wildcard subdomain; treat as leading dot "."
+                # e.g. "*.example.domain" as ".example.domain"
+                dot_no_proxy_uri = no_proxy_uri.strip("*")
```

Comment:
```suggestion
                dot_no_proxy_uri = no_proxy_uri.lstrip("*")
```
Wildcards at the end don't make much sense but we should probably avoid removing the character anyway. The code should continue to fail for malformed pattern